Lib/test/test_sys_setprofile.py

```diff
@@ -511,6 +511,26 @@ class B:
             ]
         )
 
+        # Test CALL_FUNCTION_EX
+        events = []
+        sys.setprofile(lambda frame, event, args: events.append(event))
+        # Not important, we only want to trigger INSTRUMENTED_CALL_KW
+        args = (1,)
+        B().f(*args, key=lambda x: 0)
```

Comment:
The bug is when a bound method with a non-function `im_func` attribute, right?
```suggestion
        m = B().f
        m(*args, key=lambda x: 0)
```
If we were to optimize `B().f(...)` it might not create the bound-method, leave a function/self pair on the stack. Best to protect against that.

Reply:
Yeah that's something I would not think of, but that makes sense.